tools/dump_tool/dump_tool.c

```diff
@@ -389,30 +454,97 @@ int main(int argc, char *argv[])
 		goto tty_lock_err;
 	}
 
-	if (do_handshake(dev_fd) != 0) {
-		printf("Target Handshake Failed\n");
-		goto handshake_err;
-	}
+	printf("Target device locked and ready to DUMP!!!\n");
+
+	while (1) {
+		printf("Chose from the following options:-\n1. RAM Dump\n2. Userfs Dump\n3. Both Dumps\n4. Exit Dump Tool\n5. Reboot TARGET Device\n");
+		fflush(stdout);
+		scanf("%d", &choice);
+		getchar();
+
+		switch (choice) {
+		case 2:
+			goto FS_DUMP;
+		case 4:
+			goto TOOL_EXIT;
+		case 5:
+			break;
+		case 1:
+		case 3:
+			goto RAM_DUMP;
+		default:
+			printf("Invalid Input\n");
+			continue;
+		}
 
-	printf("Target entered to ramdump mode\n");
+		printf("CONFIG_BOARD_ASSERT_AUTORESET needs to be enabled to reboot TARGET Device after a crash\n");
+		/* Handshake for synchronization before RAMDUMP */
+		strncpy(handshake_string, TARGET_REBOOT_STRING, HANDSHAKE_STR_LEN_MAX);
+		if (do_handshake(dev_fd, handshake_string) != 0) {
+			printf("Target Handshake Failed\n");
+			goto handshake_err;
+		}
+		goto TOOL_EXIT;
+
+RAM_DUMP:
+		choice--;
+		printf("DUMPING RAM CONTENTS\n");
+		fflush(stdout);
+		/* Handshake for synchronization before RAMDUMP */
+		strncpy(handshake_string, RAMDUMP_HANDSHAKE_STRING, HANDSHAKE_STR_LEN_MAX);
+		if (do_handshake(dev_fd, handshake_string) != 0) {
+			printf("Target Handshake Failed\n");
+			goto handshake_err;
+		}
 
-	if (ramdump_info_init(dev_fd) != 0) {
-		printf("Ramdump initialization failed\n");
-		goto init_err;
-	}
+		if (ramdump_info_init(dev_fd) != 0) {
+			printf("Ramdump initialization failed\n");
+			goto init_err;
+		}
 
-	if (ramdump_recv(dev_fd) != 0) {
-		printf("Ramdump receive failed\n");
-		goto ramdump_err;
+		if (ramdump_recv(dev_fd) != 0) {
+			printf("Ramdump receive failed\n");
+			goto dump_err;
+		}
+
+		printf("Ramdump received successfully..!\n");
+		fflush(stdout);
+		getchar();
+
+		/* If Userfs dump is not required, go back to input menu */
+		if (choice != 2) {
```

Comment:
> This is built in Linux so that there is no limitation. Let's use another variable to show selection. I don't want to get some affection from other code. For example, come from some value from ramdump for fsdump. I strongly want to isolate each other. I also think we don't need to use goto.

Will check with you regarding suggestions and expectations. Remaining comments updated.